metricbeat/module/system/raid/raid.go

```diff
@@ -0,0 +1,76 @@
+// +build linux
+
+package raid
+
+import (
+	"path/filepath"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/system"
+	"github.com/elastic/procfs"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	if err := mb.Registry.AddMetricSet("system", "raid", New, parse.EmptyHostParser); err != nil {
+		panic(err)
+	}
+}
+
+type MetricSet struct {
+	mb.BaseMetricSet
+	fs procfs.FS
+}
+
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Experimental("The system raid metricset is experimental")
+
+	systemModule, ok := base.Module().(*system.Module)
+	if !ok {
+		return nil, errors.New("unexpected module type")
+	}
+
+	mountPoint := filepath.Join(systemModule.HostFS, procfs.DefaultMountPoint)
+	fs, err := procfs.NewFS(mountPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	m := &MetricSet{
+		BaseMetricSet: base,
+		fs:            fs,
+	}
+
+	return m, nil
+}
+
+func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+
+	stats, err := m.fs.ParseMDStat()
+	if err != nil {
+		return nil, err
+	}
+
+	events := make([]common.MapStr, 0, len(stats))
+	for _, stat := range stats {
+		event := common.MapStr{
+			"name":           stat.Name,
+			"activity_state": stat.ActivityState,
+			"disks": common.MapStr{
+				"active": stat.DisksActive,
+				"total":  stat.DisksTotal,
```

Comment:
Usage question: How would I figure out how many available spares there are? Is `spares = total - active`? Or does `total` include the number of failed disks still in the array?

Reply:
@andrewkroh Based on https://raid.wiki.kernel.org/index.php/Mdstat I think what you state is correct.